framework/web/User.php

```diff
@@ -559,9 +593,13 @@ protected function sendIdentityCookie($identity, $duration)
      * If null, it means switching the current user to be a guest.
      * @param integer $duration number of seconds that the user can remain in logged-in status.
      * This parameter is used only when `$identity` is not null.
+     * @param boolean $cookieBased whether the login is cookie-based
      */
-    public function switchIdentity($identity, $duration = 0)
+    public function switchIdentity($identity, $duration = 0, $cookieBased = false)
```

Comment:
So wouldn’t this be OK since there is a default if the third parameter is omitted?

From: Alexander Makarov [mailto:notifications@github.com]
Sent: Monday, June 08, 2015 2:30 PM
To: yiisoft/yii2
Cc: Michael Labriola
Subject: Re: [yii2] "Move Identity Cookie code into separate functions in web\User.php" (#8698)


In framework/web/User.php<https://github.com/yiisoft/yii2/pull/8698#discussion_r31944489>:

>       */

> -    public function switchIdentity($identity, $duration = 0)

> +    public function switchIdentity($identity, $duration = 0, $cookieBased = false)

And here.

—
Reply to this email directly or view it on GitHub<https://github.com/yiisoft/yii2/pull/8698/files#r31944489>.